hydra-node/test/Hydra/Model.hs

```diff
@@ -474,11 +488,17 @@ seedWorld seedKeys seedCP = do
       pure (party, testNode, nodeThread)
     pure (res, tickThread)
 
-  modify $ \n ->
-    n
-      { nodes = Map.fromList $ (\(p, t, _) -> (p, t)) <$> fst nodes
-      , threads = snd nodes : ((\(_, _, t) -> t) <$> fst nodes)
-      }
+  s <- ask
+  lift $
+    atomically $
+      modifyTVar (nodesState s) $ \n ->
+        n
+          { nodes = Map.fromList $ (\(p, t, _) -> (p, t)) <$> fst nodes
+          , threads = snd nodes : ((\(_, _, t) -> t) <$> fst nodes)
+          }
+
+gets :: MonadSTM m => (Nodes m -> a) -> RunMonad m a
+gets f = f <$> (ask >>= lift . readTVarIO . nodesState)
```

Comment:
Seeing my comment above and this definition.

Could: define `instance MonadSTM m => MonadState (RunMonad m)`

This would yield appropriate definitions for `gets` and `modify` (besides `get` and `put`)